public/app/features/dashboard/components/AnnotationSettings/AngularEditorLoader.tsx

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useRef, useState } from 'react';
+import { AnnotationQuery, DataSourceApi } from '@grafana/data';
+import { AngularComponent, getAngularLoader } from '@grafana/runtime';
+
+export interface Props {
+  annotation: AnnotationQuery;
+  datasource: DataSourceApi;
+  onChange: (annotation: AnnotationQuery) => void;
+}
+
+export const AngularEditorLoader: React.FC<Props> = React.memo(({ annotation, datasource, onChange }) => {
+  const ref = useRef<HTMLDivElement>(null);
+  const [angularComponent, setAngularComponent] = useState<AngularComponent | null>(null);
+
+  useEffect(() => {
+    return () => {
+      if (angularComponent) {
+        angularComponent.destroy();
+      }
+    };
+  }, [angularComponent, ref]);
```

Comment:
Is the `ref` dependency necessary here?

Reply:
yea, good catch. I used ref in this one before but then moved it out to a different useEffect